module-1/src/Components/MovieListPage/MovieListPage.tsx

```diff
@@ -0,0 +1,140 @@
+import { useEffect, useMemo, useState } from "react";
+import { TestData } from "../../TestData";
+import { Genres } from "../Genres/Genres";
+import { MovieTile } from "../MovieTile/MovieTile";
+import { Search } from "../Search/Search";
+import { SortControl } from "../SortControl/SortControl";
+
+let sortValues = TestData.SearchFields.slice(0, 5);
+let listOfGenres = TestData.genres;
+const EMPTYSTRING = "";
+const MAINURL = "http://localhost:4000/movies?limit=20";
+
+export const MovieListPage = () => {
+  const [selectedGenre, setSelectedGenre] = useState(listOfGenres[0].genreName);
+  const [sortByField, setSortByField] = useState("title");
+  const [searchValue, setSearchValue] = useState("");
+  const [movies, setMovies] = useState<Movie[]>([]);
+  const [isLoading, setIsLoading] = useState(true);
+
+  useEffect(() => {
+    fetch(buildUrl)
+      .then((response) => response.json())
+      .then((result) => {
+        let res = result.data as Movie[];
+        setMovies(res);
+      })
+      .catch((e) => console.log(e))
+      .finally(() => setIsLoading(false));
```

Comment:
Reuse `fetchUrl`
```suggestion
    fetchUrl(defaultURL);
```
where `defaultURL` will be built by the same `buildURL` function with some default parameters.